lib/ocg_schemas/src/registry.rs

```diff
@@ -140,13 +141,13 @@ impl<'a> Display for RegistryNameRef<'a> {
 }
 
 /// Needs to be implemented on any object that can be a part of a Registry
-pub trait RegistryObject {
+pub trait RegistryObject: PartialEq {
     /// Should be trivial
     fn registry_name(&self) -> RegistryNameRef;
 }
 
 /// A data structure for keeping track of a stable mapping between: namespaced strings, numerical IDs and objects.
-#[derive(Serialize, Deserialize)]
+#[derive(Serialize, Deserialize, Resource)]
```

Comment:
These shouldn't be resources directly, there will be proxy objects in common that will have net sync code etc.